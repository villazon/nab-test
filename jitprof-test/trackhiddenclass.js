function C(i) {
  if (i % 2 === 0) {
    this.a = Math.random();
    this.b = Math.random();
  } else {
    this.b = Math.random();
    this.a = Math.random();
  }
}
function C2(i) {
  if (i % 2 === 0) {
    this.a = Math.random();
    this.b = Math.random();
  } else {
    this.a = Math.random();
    this.b = Math.random();
  }
}
function sum(base, p1, p2) {
  return base[p1] + base[p2];
}
for(var i=1;i<1000;i++) {
  sum(new C(i), 'a', 'b');
  sum(new C2(i), 'a', 'b');
}
