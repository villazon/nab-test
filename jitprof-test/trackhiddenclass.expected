


---------------------------
Created 9 hidden classes.
f_count: 0

property access at (jitprof-test/trackhiddenclass.js:20:10:20:18) has missed cache 1998 time(s).
  accessed property "a" of object created at undefined 1998 time(s) 
  layout [__proto__:f2|b:n|a:n|]:
	put field: a observed 500 time(s)
  layout [__proto__:f3|a:n|b:n|]:
	put field: a observed 999 time(s)
  layout [__proto__:f2|a:n|b:n|]:
	put field: a observed 499 time(s)
property access at (jitprof-test/trackhiddenclass.js:20:21:20:29) has missed cache 1998 time(s).
  accessed property "b" of object created at undefined 1998 time(s) 
  layout [__proto__:f2|b:n|a:n|]:
	put field: b observed 500 time(s)
  layout [__proto__:f3|a:n|b:n|]:
	put field: b observed 999 time(s)
  layout [__proto__:f2|a:n|b:n|]:
	put field: b observed 499 time(s)
[****]HiddenClass: 2
